File: sandbox/exercism/ruby/2024/anagram.rb
# https://exercism.org/tracks/ruby/exercises/anagram

class Anagram
  def initialize(word)
    @word = word
    @chars = word.downcase.chars.sort
  end

  def match(candidates)
    candidates.select { |c| !c.casecmp?(@word) && c.downcase.chars.sort == @chars }
  end
end

File: sandbox/exercism/go/2025/anagram.go
// https://exercism.org/tracks/go/exercises/anagram

package anagram

import (
	"sort"
	"strings"
)

func sortedString(input string) string {
	runes := []rune(strings.ToLower(input))

	sort.Slice(runes, func(x, y int) bool {
		return runes[x] < runes[y]
	})

	return string(runes)
}

func Detect(subject string, candidates []string) []string {
	result := []string{}
	sortedSubject := sortedString(subject)

	for _, candidate := range candidates {
		if (!strings.EqualFold(subject, candidate)) && (strings.EqualFold(sortedSubject, sortedString(candidate))) {
			result = append(result, candidate)
		}
	}

	return result
}
